relay-filter/src/config.rs

```diff
@@ -192,6 +196,102 @@ impl LegacyBrowsersFilterConfig {
     }
 }
 
+/// Configuration for a generic filter.
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GenericFilterConfig {
+    /// Specifies whether this filter is enabled.
+    pub is_enabled: bool,
+    /// The condition for the filter.
+    pub condition: Option<RuleCondition>,
+}
+
+impl GenericFilterConfig {
+    /// Returns true if the filter is not enabled or no condition was supplied.
+    pub fn is_empty(&self) -> bool {
+        !self.is_enabled || self.condition.is_none()
+    }
+}
+
+/// Configuration for a set of ordered filters.
+#[derive(Clone, Debug, Default)]
+pub struct OrderedFilters(pub Vec<(String, GenericFilterConfig)>);
+
+impl OrderedFilters {
+    /// Returns true if there are no generic filters configured.
+    pub fn is_empty(&self) -> bool {
+        self.0.is_empty() || self.0.iter().all(|(_, value)| value.is_empty())
+    }
+}
+
+impl Serialize for OrderedFilters {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut map = serializer.serialize_map(Some(self.0.len()))?;
+        for (filter_name, filter_config) in self.0.iter() {
+            map.serialize_entry(filter_name, filter_config)?
+        }
+        map.end()
+    }
+}
+
+struct OrderedFiltersVisitor();
```

Comment:
I think you don't need the `()` here:
```suggestion
struct OrderedFiltersVisitor;
```